framework/auth/core.py

```diff
@@ -42,17 +43,19 @@
 logger = logging.getLogger(__name__)
 
 
-# Hide implementation of token generation
-def generate_confirm_token():
-    return security.random_string(30)
-
-
-def generate_claim_token():
+# verification key v1
+def generate_verification_key():
     return security.random_string(30)
 
 
-def generate_verification_key():
-    return security.random_string(30)
+# verification key v2, expires in 30 minutes
+def generate_verification_key_v2():
```

Comment:
That could be handled by building `email_verifications` slightly differently in `add_unconfirmed_email`, i.e. the lines 
``` python
token = generate_verification_key()

# handle when email_verifications is None
if not self.email_verifications:
    self.email_verifications = {}

# confirmed used to check if link has been clicked
self.email_verifications[token] = {'email': email,
    'confirmed': False}
self._set_email_token_expiration(token, expiration=expiration)
```
could be changed to something like
``` python
token_dict = generate_verification_key(type=CONFIRM)

# handle when email_verifications is None
if not self.email_verifications:
    self.email_verifications = {}

# confirmed used to check if link has been clicked
self.email_verifications[token_dict['token']] = {
    'email': email,
    'expires': token_dict['expires'],
    'confirmed': False
}
```
which also obviates the `_set_email_token_expiration` method. Again, not a blocker, but consider adding a `TODO`